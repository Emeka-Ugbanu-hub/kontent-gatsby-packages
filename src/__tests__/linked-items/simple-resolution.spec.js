
const { KenticoCloudJsSdkTestHttpService }
  = require('kentico-cloud-js-sdk-test-http-service');

const { sourceNodes } = require('../../../gatsby-node');

// Project ID 4ac85e55-6daf-0122-7a52-81a12e5d986a
const simpleLinkedItemFakeItemsResponse =
  require('./simpleLinkedItemFakeItemsResponse.json');
const fakeTypeResponse =
  require('./fakeTypeResponse.json');

describe(
  `Simple linked items element contains reference to another items`,
  async () => {
    const fakeHttpServiceConfig = new Map();
    fakeHttpServiceConfig.set(
      /https:\/\/deliver.kenticocloud.com\/.*\/items/,
      {
        fakeResponseJson: simpleLinkedItemFakeItemsResponse,
        throwCloudError: false,
      });
    fakeHttpServiceConfig.set(
      /https:\/\/deliver.kenticocloud.com\/.*\/types/,
      {
        fakeResponseJson: fakeTypeResponse,
        throwCloudError: false,
      });

    const dummyCreateNodeID = jest.fn();
    dummyCreateNodeID.mockImplementation((input) => `dummy-${input}`);

    const dummyCreation = {
      actions: {
        createNode: jest.fn(),
      },
      createNodeId: dummyCreateNodeID,
    };
    const createNodeMock = jest.fn();
    const actions = {
      actions: {
        createNode: createNodeMock,
      },
      createNodeId: dummyCreation.createNodeId,
    };

    const deliveryClientConfig = {
      projectId: 'dummyProject',
      typeResolvers: [],
      httpService: new KenticoCloudJsSdkTestHttpService(
        fakeHttpServiceConfig
      ),
    };

    const pluginConfiguration = {
      deliveryClientConfig,
      languageCodenames: ['default'],
    };

    it('passes with no error', async () => {
      await sourceNodes(actions, pluginConfiguration);

      const calls = createNodeMock.mock.calls;
      expect(calls).toMatchSnapshot();
    });
  });
