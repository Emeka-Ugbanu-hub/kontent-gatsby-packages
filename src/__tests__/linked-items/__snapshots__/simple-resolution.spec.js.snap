// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple linked items element contains reference to another items passes with no error 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-cloud-item-main-project-default",
        "dummy-kentico-cloud-item-sub-project-1-default",
        "dummy-kentico-cloud-item-sub-project-2-default",
      ],
      "elements": Array [
        Object {
          "codename": "related_projects",
          "name": "Related projects",
          "options": Array [],
          "type": "modular_content",
        },
        Object {
          "codename": "content",
          "name": "Content",
          "options": Array [],
          "type": "text",
        },
      ],
      "id": "dummy-kentico-cloud-type-project",
      "internal": Object {
        "content": "{\\"system\\":{\\"codename\\":\\"project\\",\\"id\\":\\"ec706130-b0dd-401c-ae1a-10f2eeb2eb89\\",\\"name\\":\\"Project\\",\\"lastModified\\":\\"2018-12-28T12:29:09.4933517Z\\"},\\"elements\\":[{\\"options\\":[],\\"codename\\":\\"related_projects\\",\\"name\\":\\"Related projects\\",\\"type\\":\\"modular_content\\"},{\\"options\\":[],\\"codename\\":\\"content\\",\\"name\\":\\"Content\\",\\"type\\":\\"text\\"}]}",
        "contentDigest": "98b4a5274bd4a532bf161ca1f3631df1",
        "type": "KontentTypeProject",
      },
      "parent": null,
      "system": Object {
        "codename": "project",
        "id": "ec706130-b0dd-401c-ae1a-10f2eeb2eb89",
        "lastModified": "2018-12-28T12:29:09.4933517Z",
        "name": "Project",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-cloud-type-project",
      "elements": Object {
        "content": Object {
          "name": "Content",
          "type": "text",
          "value": "Main project content",
        },
        "related_projects": Object {
          "itemCodenames": Array [
            "sub_project_1",
            "sub_project_2",
          ],
          "linked_items___NODE": Array [
            "dummy-kentico-cloud-item-sub-project-1-default",
            "dummy-kentico-cloud-item-sub-project-2-default",
          ],
          "name": "Related projects",
          "type": "modular_content",
        },
      },
      "id": "dummy-kentico-cloud-item-main-project-default",
      "internal": Object {
        "content": "{\\"system\\":{\\"name\\":\\"Main project\\",\\"codename\\":\\"main_project\\",\\"id\\":\\"1a3aa6bd-7b4b-486f-86ab-7bfe359ad614\\",\\"lastModified\\":\\"2018-12-28T12:53:59.075Z\\",\\"language\\":\\"default\\",\\"type\\":\\"project\\",\\"sitemapLocations\\":[]},\\"elements\\":{\\"content\\":{\\"name\\":\\"Content\\",\\"type\\":\\"text\\",\\"value\\":\\"Main project content\\"},\\"related_projects\\":{\\"name\\":\\"Related projects\\",\\"type\\":\\"modular_content\\",\\"itemCodenames\\":[\\"sub_project_1\\",\\"sub_project_2\\"]}},\\"preferred_language\\":\\"default\\"}",
        "contentDigest": "ffad216a3735a2333747cb85ec66cef6",
        "type": "KontentItemProject",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "main_project",
        "id": "1a3aa6bd-7b4b-486f-86ab-7bfe359ad614",
        "language": "default",
        "lastModified": "2018-12-28T12:53:59.075Z",
        "name": "Main project",
        "sitemapLocations": Array [],
        "type": "project",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-cloud-type-project",
      "elements": Object {
        "content": Object {
          "name": "Content",
          "type": "text",
          "value": "Sub project 1 content",
        },
        "related_projects": Object {
          "itemCodenames": Array [],
          "linked_items___NODE": Array [],
          "name": "Related projects",
          "type": "modular_content",
        },
      },
      "id": "dummy-kentico-cloud-item-sub-project-1-default",
      "internal": Object {
        "content": "{\\"system\\":{\\"name\\":\\"Sub project 1\\",\\"codename\\":\\"sub_project_1\\",\\"id\\":\\"8a7f275d-0e1c-4b30-af3e-f02c91259faf\\",\\"lastModified\\":\\"2019-01-14T15:52:54.904Z\\",\\"language\\":\\"default\\",\\"type\\":\\"project\\",\\"sitemapLocations\\":[]},\\"elements\\":{\\"content\\":{\\"name\\":\\"Content\\",\\"type\\":\\"text\\",\\"value\\":\\"Sub project 1 content\\"},\\"related_projects\\":{\\"name\\":\\"Related projects\\",\\"type\\":\\"modular_content\\",\\"itemCodenames\\":[]}},\\"preferred_language\\":\\"default\\"}",
        "contentDigest": "1ed4ffc99a2861d499cdb18bd8a301b0",
        "type": "KontentItemProject",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "sub_project_1",
        "id": "8a7f275d-0e1c-4b30-af3e-f02c91259faf",
        "language": "default",
        "lastModified": "2019-01-14T15:52:54.904Z",
        "name": "Sub project 1",
        "sitemapLocations": Array [],
        "type": "project",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-cloud-item-main-project-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-cloud-type-project",
      "elements": Object {
        "content": Object {
          "name": "Content",
          "type": "text",
          "value": "Sub project 2 content",
        },
        "related_projects": Object {
          "itemCodenames": Array [],
          "linked_items___NODE": Array [],
          "name": "Related projects",
          "type": "modular_content",
        },
      },
      "id": "dummy-kentico-cloud-item-sub-project-2-default",
      "internal": Object {
        "content": "{\\"system\\":{\\"name\\":\\"Sub project 2\\",\\"codename\\":\\"sub_project_2\\",\\"id\\":\\"91703600-4d43-410e-ba48-93c6c8b0a754\\",\\"lastModified\\":\\"2018-12-28T12:30:21.664Z\\",\\"language\\":\\"default\\",\\"type\\":\\"project\\",\\"sitemapLocations\\":[]},\\"elements\\":{\\"content\\":{\\"name\\":\\"Content\\",\\"type\\":\\"text\\",\\"value\\":\\"Sub project 2 content\\"},\\"related_projects\\":{\\"name\\":\\"Related projects\\",\\"type\\":\\"modular_content\\",\\"itemCodenames\\":[]}},\\"preferred_language\\":\\"default\\"}",
        "contentDigest": "01e161a742aa48aaad251f263b292722",
        "type": "KontentItemProject",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "sub_project_2",
        "id": "91703600-4d43-410e-ba48-93c6c8b0a754",
        "language": "default",
        "lastModified": "2018-12-28T12:30:21.664Z",
        "name": "Sub project 2",
        "sitemapLocations": Array [],
        "type": "project",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-cloud-item-main-project-default",
      ],
    },
  ],
]
`;
